src/solvers/dg_tree/dg_1d.jl

```diff
@@ -342,13 +360,13 @@ end
 
 # TODO: Taal dimension agnostic
 function calc_boundary_flux!(cache, t, boundary_condition::BoundaryConditionPeriodic,
-                             equations::AbstractEquations{1}, dg::DG)
+                             mesh::TreeMesh, equations::AbstractEquations{1}, dg::DG)
```

Comment:
Sometimes you dispatch on `::TreeMesh`, sometimes not... What's the rule/heuristic you're using here?

Reply:
The `CurvedMesh` (aka `StructuredMesh`) uses the methods where I don't dispatch on `TreeMesh`... We should discuss whether we want to dispatch explicitly on `Union{TreeMesh, CurvedMesh}`, on no specific type, on some supertype, ... and where we want to collect methods like this

Reply:
Ah ok. Well, in general I'd say "without annotation is better than with annotation" where it still works, thus I'm fine with the way it currently is implemented. That way, other DGSEM mesh types can just reuse this if the shoe fits.

IMHO #536 is really the place to continue this discussion, so I'll close this here.